src/Servers/Kestrel/Kestrel/test/ConfigurationReaderTests.cs

```diff
@@ -173,5 +174,70 @@ public void ReadEndpointsSection_ReturnsCollection()
             Assert.Equal("cetlocation", cert4.Location);
             Assert.True(cert4.AllowInvalid);
         }
+
+        [Fact]
+        public void ReadEndpointWithSingleSslProtocolSet_ReturnsCorrectValue()
+        {
+            var config = new ConfigurationBuilder().AddInMemoryCollection(new[]
+            {
+                new KeyValuePair<string, string>("Endpoints:End1:Url", "http://*:5001"),
+                new KeyValuePair<string, string>("Endpoints:End1:SslProtocols:0", "Tls11"),
+            }).Build();
+            var reader = new ConfigurationReader(config);
+
+            var endpoint = reader.Endpoints.First();
+            Assert.Equal(SslProtocols.Tls11, endpoint.SslProtocols);
+        }
+
+        [Fact]
+        public void ReadEndpointWithMultipleSslProtocolsSet_ReturnsCorrectValue()
+        {
+            var config = new ConfigurationBuilder().AddInMemoryCollection(new[]
+            {
+                new KeyValuePair<string, string>("Endpoints:End1:Url", "http://*:5001"),
+                new KeyValuePair<string, string>("Endpoints:End1:SslProtocols:0", "Tls11"),
+                new KeyValuePair<string, string>("Endpoints:End1:SslProtocols:1", "Tls12"),
```

Comment:
Just confirming the case insensitivity.
```suggestion
                new KeyValuePair<string, string>("Endpoints:End1:SslProtocols:1", "TLS12"),
```

Reply:
Makes sense, added a separate test for it - https://github.com/dotnet/aspnetcore/pull/22910/commits/854555292407266e94396a90e9cfad336374f7c7#diff-f2012370fc7879ce8c37dfb628b973beR207